test/integration/protocol_integration_test.cc

```diff
@@ -3974,21 +4011,79 @@ TEST_P(DownstreamProtocolIntegrationTest, InvalidRequestHeaderName) {
       codec_client_->startRequest(Http::TestRequestHeaderMapImpl{{":method", "POST"},
                                                                  {":path", "/test/long/url"},
                                                                  {":authority", "sni.lyft.com"},
+                                                                 {":scheme", "http"},
                                                                  {"foo}name", "foo_value"}});
   auto response = std::move(encoder_decoder.second);
 
+  ASSERT_TRUE(codec_client_->waitForDisconnect());
```

Comment:
was this failing on wrong error before?

Reply:
It was not checking for client disconnect for H/2 case, but waiting for reset with the `ConnectionTermination` reason with is equivalent. I have simply harmonized the condition that test test should be waiting on.